### DISCUSSION TITLE
SDK missing primary functions for SPI such as data transfer/receive and setting clock speed

### DISCUSSION THREAD

**Discussion by XanCraft21** [espressif/arduino-esp32#7880] [General]
Hello, i am having trouble with the sdk. There is no issue template specifically for SDK issues, so i had to do it here. I just discovered that the SDK is missing important SPI driver functions. I don’t see a way to change settings like clock speed and spi mode or any way to send or receive data. I know the arduino SPI library is a thing but i need to configure setting that can only be found in the SDK. It doesn’t help that the SDK is over complicated. If anyone can tell me if there is a functions i can use, or how to make my own function, or if a dev could add those functions in. Thank you.

EDIT: I forgot to mention there is a way to transmit and receive but it’s too complex and can’t be used easily.

**Comment by me-no-dev** [comment]
https://github.com/espressif/arduino-esp32/blob/master/libraries/SPI/src/SPI.h#L63-L70

SPI is not missing any of the functions you need. Did you look at the wrong place?

**Reply by XanCraft21** [reply]
Maybe i did look in the wrong please, but i was looking for the espressif sdk version in the core package. I configured my pins i want to use like for quad or octal spi and i can init the bus, but i can’t change any other settings or send data. I know how to use the original spi control with SPIClass but i wanted to try something new. Looking at the official website for the sdk does not really tell how to use it very well, all there is is an example, but it does not set the clock speed, and how is the data being transfered?

could it be that both the sdk spi and arduino spi are connected together and work with each other?

EDIT: I looked more at the example code i found and it seems to be super complex. I really hope that both spi types are combined somehow.
